### ISSUE TITLE
Add support for state diagrams

### ISSUE THREAD

**Issue by knsv** [knsv/mermaid#945]
A new diagram type similar to the state diagrams in plantUML.

```
state
   [*] --> State1
   State1 --> [*]
   State1 : this is a string
   State1 : this is another string
   
   State1 -> State2
   State2 --> [*]
```

**Comment by nothingismagick** [comment]
Love this idea.

**Comment by barakplasma** [comment]
[Link to PlantUML State Diagrams page](http://plantuml.com/state-diagram)
![Screen Shot 2019-09-25 at 9 55 53](https://user-images.githubusercontent.com/62937/65576465-b0662880-df7a-11e9-8869-56b1040e2946.png)

**Comment by Dunning-Kruger** [comment]
This was released in 8.4.0. Closing soon, if no objections.
